### PR TITLE
Separated Parser and Connection Process

### PR DESCRIPTION
**Pull request by ConorPKeegan**
# Changes
* Separated the KiVi Walker Parser from Connection Proxy
* Created a Base class for the Parser
* Created a factory for the parser
* Separated the Connection Process from the Connection Proxy
* Created a factory for the creation, and starting of the Connection Process
* Added unit tests for the factories, and the Parser Base class

### CODE REVIEW

**Comment by ConorPKeegan** [comment]
@PrzemekWirkus

File: mbed_host_tests/host_tests_conn_process/conn_process.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from multiprocessing import Process
+from mbed_host_tests.host_tests_conn_process import run_default_conn_process
+
+def conn_process_factory(logger, options, mbed, event_queue, dut_event_queue, conn_process_name="greentea-client"):
```
Comment (PrzemekWirkus) [inline mbed_host_tests/host_tests_conn_process/conn_process.py]: We need there doxygen comments for each method/function :)

File: mbed_host_tests/host_tests_conn_process/default_conn_process.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+"""
+
+import uuid
+from time import time
+from Queue import Empty as QueueEmpty
+from mbed_host_tests.host_tests_logger import HtrunLogger
+from mbed_host_tests.host_tests_parser import parser_factory
+from mbed_host_tests.host_tests_conn_proxy import conn_primitive_factory
+
+def run_default_conn_process(
```
Comment (PrzemekWirkus) [inline mbed_host_tests/host_tests_conn_process/default_conn_process.py]: We need there doxygen comments for each method/function :)

**Comment by PrzemekWirkus** [comment]
I will merge after few simple comments that are missing :)

**Comment by PrzemekWirkus** [comment]
+1

**Comment by PrzemekWirkus** [comment]
I will retest this :)

**Pull request merged**